go-gitea/gitea
# Show information about a topic

This shows some Information about a Topic, when a User visits it. It uses the same Data as GitHub.

As you can see, this PR adds a lot of data files. They are all taken from [github/explore](https://github.com/github/explore).

I don't know, if the Maintainers really wants this, as it adds a lot of files for such a small feature. Therefore I decided to just code a first version and create a draft PR. This PR is more like a Proof of Concept than finished Code. It was programmed Quick and Dirty. I don't like putting too much effort into something that won't be merged, so I decided to create this and see, if it will be accepted. If yes, I will continue working on it. If no, not much work is lost.

**Why I put this into the Database instead of reading the files from disk?**
The Data contains aliases. Take for example this from the `3D`:
```yaml
aliases: 3d-printing, 3d-graphics, 3d-modeling, 3d-animation, 3d-cad, 3d-rendering, 3d-models
```
This means the Data for this Topic is also for all those topics. Instead of creating a lookup table, i decided to put all the Data in the Database to make it easier. That will probably change in the Future.

**How to test this?**
Go to the Admin Dashboard and run "Update topics". After that, you should see the Information when visiting a Topic.

**Screenshot**
This is the result of 5 minutes frontend work:
![grafik](https://github.com/go-gitea/gitea/assets/15185051/40f80448-ceeb-443b-9e66-f22aa26968d0)
[Here is the same on GitHub as comparison](https://github.com/topics/git)

**This Code is currently not in a state, where it's ready for Code reviews, so at the Moment please discuss only if you want this or not. You can also discuss about some implementation concepts. If the Maintainers want this feature, I will actually working on this.**
